src/vea/vea_free.c

```diff
@@ -344,20 +473,241 @@ merge_free_ext(struct vea_space_info *vsi, struct vea_free_extent *ext_in,
 
 	if (type == VEA_TYPE_AGGREGATE || type == VEA_TYPE_COMPOUND) {
 		neighbor->vfe_age = merged.vfe_age;
-		rc = dock_entry(vsi, neighbor_entry, type);
+		rc = dock_extent_entry(vsi, neighbor_entry, type);
 		if (rc < 0)
 			return rc;
 	}
 
 	return 1;
 }
 
-/* Free extent to in-memory compound index */
+static inline int
+entry_adjacent(struct vea_free_entry *entry1, struct vea_free_entry *entry2)
+{
+	if (entry1->vfe_type != entry2->vfe_type ||
+	    entry1->vfe_class != entry2->vfe_class)
+		return 0;
+
+	return ext_adjacent(&entry1->vfe_ext, &entry2->vfe_ext);
+}
+
+
+/*
+ * Make sure there is no overlapping or duplicated extents in the free
+ * extent tree. The passed in @ext_in will be merged with adjacent extents
+ * and being inserted in the tree.
+ *
+ * Return value:	0	- Not merged
+ *			1	- Merged in tree
+ *			-ve	- Error
+ */
+static int
+merge_free_entry(struct vea_space_info *vsi, struct vea_free_entry *entry_in,
+		 unsigned int flags)
+{
```

Comment:
Looks dup of merge_free_ext(), if we use per bitmap chunk aggregate tree, we could just reuse the merge_free_ext()?